flocker/node/agents/ebs.py

```diff
@@ -224,4 +409,24 @@ def destroy_volume(self, blockdevice_id):
         raise UnknownVolume(blockdevice_id)
 
     def get_device_path(self, blockdevice_id):
-        pass
+        """
+        Get device path for the EBS volume corresponding to the given
+        block device.
+
+        :param unicode blockdevice_id: EBS UUID for the volume to look up.
+
+        :returns: A ``FilePath`` for the device.
+        :raises UnknownVolume: If the supplied ``blockdevice_id`` does not
+            exist.
+        :raises UnattachedVolume: If the supplied ``blockdevice_id`` is
+            not attached to a host.
+        """
+        volume = self._get(blockdevice_id)
+        if volume.attached_to is None:
+            raise UnattachedVolume(blockdevice_id)
+
+        ebs_volume = self._get_ebs_volume(blockdevice_id)
+        device = ebs_volume.tags[ATTACHED_DEVICE_LABEL]
+        if device is None:
+            raise UnattachedVolume(blockdevice_id)
```

Comment:
Good catch! Added test to demonstrate need to handle KeyError:
===============================================================================
[ERROR]
Traceback (most recent call last):
  File "/home/ubuntu/flocker/.tox/py27/local/lib/python2.7/site-packages/flocker/node/agents/functional/test_ebs.py", line 133, in test_attached_volume_missing_device_tag
    device = self.api.get_device_path(volume.blockdevice_id)
  File "/home/ubuntu/flocker/.tox/py27/local/lib/python2.7/site-packages/flocker/node/agents/ebs.py", line 442, in get_device_path
    device = ebs_volume.tags[ATTACHED_DEVICE_LABEL]
exceptions.KeyError: u'attached-device-name'

flocker.node.agents.functional.test_ebs.EBSBlockDeviceAPIInterfaceTests.test_attached_volume_missing_device_tag
-------------------------------------------------------------------------------